code-challenges.test.js

```diff
@@ -14,38 +14,141 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const people = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
+
+
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
+describe("createSentence", () => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", ()=> {
+    const people = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+      ]
+      expect(createSentence(people)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+      
+  })
+})
+
+// ReferenceError: createSentence is not defined
 
 // b) Create the function that makes the test pass.
 
+// Define a function called createSentence.
+// Set parameter to peopleArray.
+// Use .map to iterate over the array. 
+//Use the .split method to separarte first and last name and set this to a variable called name.
+// Create a variable called firstLetter set equal to name[0] then use the .slice method from 0,1 with .toUpperCase to capitalize the first letter of the first name.
+//Create a firstName variable set equal to name[0].slice(1) to return the rest of the letters of the first name.
+// Create a lastLetter variable equal to name[1] and repeat above step to capitalize the first letter of the last name.
+//Create a lastName variable set equal to name[1].slice(1) to return the rest of the letters of the last name.
+//Create a variable called occupation set equal to the occupation in the array.
+//Return first name, last name is occupation using string interpolation.
+
+const createSentence = (peopleArray) => {
+  return peopleArray.map(person => {
+   let name = person.name.split(" ")
+   let firstLetter = name[0].slice(0, 1).toUpperCase()
+   let firstName = name[0].slice(1)
+   let lastLetter = name[1].slice(0, 1).toUpperCase()
+   let lastName = name[1].slice(1)
+   let occupation = person.occupation
+   return `${firstLetter}${firstName} ${lastLetter}${lastName} is ${occupation}.`
+  })
+}
```

Comment:
Nice work!  This problem is super challenging, but you got it! Your variable names are super descriptive, which makes this very easy to read. If you want to stretch yourself, try chaining and combining some of these variables and methods together.